tests/acceptance/core/service/customflag/model.go

```diff
@@ -0,0 +1,141 @@
+package customflag
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+var ServiceFlag FlagConfig
+
+type FlagConfig struct {
+	InstallType       InstallTypeValueFlag
+	InstallUpgrade    MultiValueFlag
+	TestCase          TestConfigFlag
+	ClusterConfig     ClusterConfigFlag
+	UpgradeVersionSUC UpgradeVersionFlag
+}
+
+// UpgradeVersionFlag is a custom type to use upgradeVersionSUC flag
+type UpgradeVersionFlag struct {
+	Version string
+}
+
+// InstallTypeValueFlag is a customflag type that can be used to parse the installation type
+type InstallTypeValueFlag struct {
+	Version string
+	Commit  string
+}
+
+// TestConfigFlag is a customflag type that can be used to parse the test case argument
+type TestConfigFlag struct {
+	TestFuncName   *string
+	TestFunc       TestCaseFlagType
+	DeployWorkload bool
+}
+
+// TestCaseFlagType is a customflag type that can be used to parse the test case argument
+type TestCaseFlagType func(deployWorkload bool)
+
+// MultiValueFlag is a customflag type that can be used to parse multiple values
+type MultiValueFlag []string
+
+type DestroyFlag bool
+
+// ClusterConfigFlag is a customFlag type that can be used to change some cluster config
+type ClusterConfigFlag struct {
+	Destroy DestroyFlag
+}
+
+// String returns the string representation of the MultiValueFlag
+func (m *MultiValueFlag) String() string {
+	return strings.Join(*m, ",")
+}
+
+// Set func sets multiValueFlag appending the value
+func (m *MultiValueFlag) Set(value string) error {
+	*m = append(*m, value)
+	return nil
+}
+
+// String returns the string representation of the TestConfigFlag
+func (t *TestConfigFlag) String() string {
+	return fmt.Sprintf("TestFuncName: %s, DeployWorkload: %t", t.TestFuncName, t.DeployWorkload)
+}
+
+// Set parses the customflag value for TestConfigFlag
+func (t *TestConfigFlag) Set(value string) error {
+	parts := strings.Split(value, ",")
+
+	if len(parts) < 1 {
+		return fmt.Errorf("invalid test case customflag format")
+	}
+
+	t.TestFuncName = &parts[0]
+	if len(parts) > 1 {
+		deployWorkload, err := strconv.ParseBool(parts[1])
+		if err != nil {
+			return fmt.Errorf("invalid deploy workload customflag: %v", err)
+		}
+		t.DeployWorkload = deployWorkload
+	}
+
+	return nil
+}
+
+// String returns the string representation of the InstallTypeValue
+func (i *InstallTypeValueFlag) String() string {
+	return fmt.Sprintf("Version: %s, Commit: %s", i.Version, i.Commit)
+}
+
+// Set parses the input string and sets the Version or Commit field using Set customflag interface
+func (i *InstallTypeValueFlag) Set(value string) error {
+	parts := strings.Split(value, "=")
+
+	if len(parts) == 2 {
+		switch parts[0] {
+		case "INSTALL_RKE2_VERSION":
```

Comment:
We may need INSTALL_RKE2_CHANNEL here for rpm installs

Reply:
done